src/sentry/rules/actions/base.py

```diff
@@ -60,6 +63,39 @@ def after(self, event: Event, state: EventState) -> Generator[CallbackFuture, No
         pass
 
 
+class SentryAppEventAction(EventAction, abc.ABC):
+    """Abstract class to ensure that actions in SENTRY_APP_ACTIONS have all required methods"""
+
+    @abc.abstractmethod
+    def get_custom_actions(self, project: Project) -> Sequence[Mapping[str, Any]]:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def self_validate(self) -> None:
+        raise NotImplementedError()
+
+
+def trigger_sentry_app_action_creators_for_issues(
```

Comment:
This is for issue alerts, we need to fire the requesters if any SentryAppActions are present. The equivalent for metric alerts is in `incidents/logic.py`